project_1/include/server.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "packet.hpp"
+
+#include <cstdint>
+#include <vector>
+#include <queue>
+
+// Type for server receiver, contains only ID
+struct ServerBase {
+protected:
+    std::uint64_t id;
+public:
+    ServerBase(std::uint32_t);
+    ServerBase() = default;
+
+    std::uint32_t GetID() const;
+};
+
+// Type for sending server, contains ID and distance to receiver
+struct ServerSender : ServerBase {
```

Comment:
`struct` members are public by default
Please, also follow the declaration order https://google.github.io/styleguide/cppguide.html#Declaration_Order